browser_history/generic.py

```diff
@@ -175,56 +189,95 @@ def fetch(self, history_paths=None, sort=True, desc=False):
+            data member entries set to list(tuple(:py:class:`datetime.datetime`, str))
+        :rtype: :py:class:`browser_history.generic.Outputs`
+        """
+
+        if bookmarks_paths is None:
+            bookmarks_paths = self.paths(profile_file=self.bookmarks_file)
+        output_object = Outputs(fetch_type="bookmarks")
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            for bookmarks_path in bookmarks_paths:
+                if not os.path.exists(bookmarks_path):
+                    continue
+                copied_bookmark_path = shutil.copy2(
+                    bookmarks_path.absolute(), tmpdirname
+                )
+                date_bookmarks = self.bookmarks_parser(copied_bookmark_path)
+                output_object.bookmarks.extend(date_bookmarks)
+                if sort:
+                    output_object.bookmarks.sort(reverse=desc)
         return output_object
 
 
 class Outputs:
     """
-    A generic class to encapsulate history outputs and to
+    A generic class to encapsulate history and bookmark outputs and to
     easily convert them to JSON, CSV or other formats.
 
-    * **entries**: List of tuples of Timestamp & URL
-    :type entries: list(tuple(:py:class:`datetime.datetime`, str))
+    * **histories**: List of tuples of Timestamp & URL
+    :type history_entries: list(tuple(:py:class:`datetime.datetime`, str))
 
-    * **formats**: A tuple of strings containing all supported formats
+    * **bookmarks** List of tuples of Timestamp , URL , Title , Folder
+    :type bookmark_entries: list(tuple(:py:class:`datetime.datetime`, str,str,str))
 
-    * **fields**: The fields available for the history data returned
+    * **formats**: A tuple of strings containing all supported formats
 
     """
 
     # All formats added here should be implemented in _format_map
     # Formats added here and in _format_map should be in lowercase
     formats = ("csv", "json")
-  
+
     # Use the below fields for all formatter implementations
-    fields = ("Timestamp", "URL")
 
-    def __init__(self):
-        self.entries = []
+    def __init__(self, fetch_type):
+        self.histories = []
+        self.bookmarks = []
         # format map is used by the formatted method to call the right formatter
         self._format_map = {
             "csv": self.to_csv,
             "json": self.to_json,
-            "jsonl": lambda: self.to_json(json_lines=True)
+            "jsonl": lambda: self.to_json(json_lines=True),
         }
 
-    def get(self):
-        """
-        Return the list of tuples of Timestamps & URLs.
-        :rtype: list(tuple(:py:class:`datetime.datetime`, str))
-        """
+        self.fetch_type = fetch_type
 
-        return self.entries
+        self._fetch_map = {
+            "history": [self.histories, ("Timestamp", "URL")],
+            "bookmarks": [self.bookmarks, ("Timestamp", "URL", "Title", "Folder")],
+        }
```

Comment:
You could rename this to something else since `_fetch_map` is used in the CLI. maybe `_field_map`?

Reply:
Okay